src/api/controllers/sitemap/index.js

```diff
@@ -62,39 +64,68 @@ const publicRoutes = [{
   priority: '0.4',
 }];
 
+type ChangeFrequency = 'always' | 'hourly' | 'daily' | 'weekly' | 'monthly' | 'yearly' | 'never';
 type BuildUrlTemplateOptions = {
   loc: string,
   priority: string,
   updatedAt?: Date,
+  changefreq?: ChangeFrequency,
 };
 
-const buildUrlTemplate = ({ loc, updatedAt, priority }: BuildUrlTemplateOptions) => {
+const buildUrlTemplate = ({ loc, updatedAt, priority, changefreq }: BuildUrlTemplateOptions) => {
   const tags = [
-    `<loc>${config.web.publicUrl}/${loc}</loc>`,
+    `<loc>${config.web.publicUrl}/${encodeURI(loc)}</loc>`,
     updatedAt && `<lastmod>${updatedAt.toISOString()}</lastmod>`,
     priority && `<priority>${priority}</priority>`,
+    changefreq && `<changefreq>${changefreq}</changefreq>`,
   ].filter((tag) => tag);
 
   return `  <url>
     ${tags.join('\n    ')}
   </url>`;
 };
 
-const buildSitemap = (publicUpdatedMap, items) =>
+const buildSitemap = (items) =>
 `<?xml version="1.0" encoding="UTF-8"?>
 <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
-${publicRoutes.map((route) => buildUrlTemplate({
-  ...route,
-  // $FlowFixMe
-  updatedAt: publicUpdatedMap[route.loc],
-})).join('\n')}
-${items.reduce((str, item) => {
-  // eslint-disable-next-line no-param-reassign
-  str += `${item.join('\n')}\n`;
-  return str;
-}, '')}
+${items.join('\n')}
 </urlset>`;
 
+const buildSitemapData = (users, guilds, characters, publicUpdatedAtMap) => {
+  return [
```

Comment:
You can do a `[].concat(x.map(), y.map())` so you don't have to expand every list